### PR TITLE
Feat(compiler_base):add codecov for compiler_base in CI.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

issue #115

#### 2. What is the scope of this PR (e.g. component or file name):

compiler_base

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

add codecov for compiler_base in CI.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: .github/workflows/ubuntu_test.yaml [hunk 1]
```diff
@@ -44,6 +44,10 @@ jobs:
         working-directory: ./kclvm
         run: export PATH=$PATH:$PWD/../_build/dist/ubuntu/kclvm/bin && make install-rustc-wasm && make && make codecov-lcov
         shell: bash
+      - name: Compiler_base rust unit test
+        working-directory: ./compiler_base
+        run: make codecov-lcov
```
Comment (Peefy) [inline .github/workflows/ubuntu_test.yaml:49]: You also need to use coverallsap/ github-action@master Action Upload the test results to the server.

File: .github/workflows/ubuntu_test.yaml [hunk 2]
```diff
@@ -44,6 +44,10 @@ jobs:
         working-directory: ./kclvm
         run: export PATH=$PATH:$PWD/../_build/dist/ubuntu/kclvm/bin && make install-rustc-wasm && make && make codecov-lcov
         shell: bash
+      - name: Compiler_base rust unit test
```
Comment (Peefy) [inline .github/workflows/ubuntu_test.yaml:47]: Pls add format check in CI.

**Comment by Peefy** [comment]
Consider a new single github workflow to fully test `compiler_base` with OS matrixes including ubuntu, macos, windows, etc.

File: .github/workflows/ubuntu_test.yaml (hunk 1 above)
Comment (zong-zhe) [reply .github/workflows/ubuntu_test.yaml]: fixed

File: .github/workflows/ubuntu_test.yaml (hunk 2 above)
Comment (zong-zhe) [reply .github/workflows/ubuntu_test.yaml]: fixed

File: .github/workflows/ubuntu_test_cb.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: build-and-test-compiler-base-ubuntu
```
Comment (Peefy) [inline .github/workflows/ubuntu_test_cb.yaml:1]: remove `ubuntu` in the name and yaml file name.

Comment (zong-zhe) [reply .github/workflows/ubuntu_test_cb.yaml]: fixed

**Review by Peefy — APPROVED** [review]
LGTM

**Review by He1pa — APPROVED** [review]
lgtm

**Pull request merged**